include/task.h

```diff
@@ -12,54 +13,77 @@
+
 		std::pair<buffer_id, access_mode> get_nth_access(const size_t n) const {
-			const auto& [bid, rm] = m_accesses[n];
-			return {bid, rm->get_access_mode()};
+			const auto& [bid, mode, _] = m_accesses[n];
+			return {bid, mode};
 		}
 
-		/**
-		 * @brief Computes the combined access-region for a given buffer, mode and subrange.
-		 *
-		 * @param bid
-		 * @param mode
-		 * @param sr The subrange to be passed to the range mappers (extended to a chunk using the global size of the task)
-		 *
-		 * @returns The region obtained by merging the results of all range-mappers for this buffer and mode
-		 */
-		region<3> get_mode_requirements(
-		    const buffer_id bid, const access_mode mode, const int kernel_dims, const subrange<3>& sr, const range<3>& global_size) const;
-
-		region<3> get_requirements_for_nth_access(const size_t n, const int kernel_dims, const subrange<3>& sr, const range<3>& global_size) const;
-
-		std::vector<const range_mapper_base*> get_range_mappers(const buffer_id bid) const {
-			std::vector<const range_mapper_base*> rms;
-			for(const auto& [a_bid, a_rm] : m_accesses) {
-				if(a_bid == bid) { rms.push_back(a_rm.get()); }
-			}
-			return rms;
+		region<3> get_requirements_for_nth_access(const size_t n, const box<3>& execution_range) const;
+
+		/// Returns the union of all consumer accesses made across the entire task (conceptually, the
+		/// union of the set of regions obtained by calling get_consumed_region for each chunk).
+		region<3> get_task_consumed_region(const buffer_id bid) const {
+			if(auto it = m_task_consumed_regions.find(bid); it != m_task_consumed_regions.end()) { return it->second; }
+			return {};
 		}
 
-		box_vector<3> get_required_contiguous_boxes(const buffer_id bid, const int kernel_dims, const subrange<3>& sr, const range<3>& global_size) const;
+		/// Returns the union of all producer accesses made across the entire task (conceptually, the
+		/// union of the set of regions obtained by calling get_produced_region for each chunk).
+		region<3> get_task_produced_region(const buffer_id bid) const {
+			if(auto it = m_task_produced_regions.find(bid); it != m_task_produced_regions.end()) { return it->second; }
+			return {};
+		};
+
+		/// Computes the union of all consumed regions (across multiple accesses) for a given execution range.
+		region<3> compute_consumed_region(const buffer_id bid, const box<3>& execution_range) const;
+
+		/// Computes the union of all produced regions (across multiple accesses) for a given execution range.
+		region<3> compute_produced_region(const buffer_id bid, const box<3>& execution_range) const;
+
+		/// Returns a set of bounding boxes, one for each accessed region, that must be allocated contiguously.
+		box_vector<3> compute_required_contiguous_boxes(const buffer_id bid, const box<3>& execution_range) const;
 
 	  private:
-		std::vector<std::pair<buffer_id, std::unique_ptr<range_mapper_base>>> m_accesses;
+		std::vector<buffer_access> m_accesses;
```

Comment:
`std::vector<buffer_access>` is used here and also in some signatures.  
Maybe we should have an alias for it so that it's also easier to evaluate potential replacements